src/ui/FieldValue/FieldValue.ts

```diff
@@ -403,14 +403,16 @@ export class FieldValue extends Component {
 
     if (atLeastOneFacetIsEnabled) {
       const selected = find(facets, (facet: Facet) => {
-        const facetValue = facet.values.get(value);
+        const facetValue = facet.values.get(originalFacetValue);
         return facetValue && facetValue.selected;
       });
 
+      const label = selected ? l('RemoveFilterOn', renderedFacetValue) : l('FilterOn', renderedFacetValue);
       new AccessibleButton()
-        .withLabel(selected ? l('RemoveFilterOn', value) : l('FilterOn', value))
+        .withLabel(label)
+        .withTitle(label)
         .withElement(element)
-        .withSelectAction(() => this.handleSelection(selected as Facet, facets as Facet[], value))
+        .withSelectAction(() => this.handleSelection(selected as Facet, facets as Facet[], originalFacetValue))
```

Comment:
It may be useful to move the casts on `selected` and `facets` to where the variables are initialised. That way type-checks would apply to wherever they are used in the method.